FullStoryMiddleware/FSSuffixedProperties.m

```diff
@@ -110,16 +110,17 @@ - (void)addSimpleObject:(NSObject *)obj withKey:(NSString *)key {
 - (NSString *)getSuffixStringFromSimpleObject:(NSObject *)obj {
     // default to no suffix;
     NSString * suffix = @"";
+
     if ([obj isKindOfClass:[NSNumber class]]) {
         // defaut to real
         suffix = @"_real";
         NSNumber *n = (NSNumber *) obj;
         const char *typeCode = n.objCType;
-        if (*typeCode == 'i') {
-            suffix = @"_int";
-        } else if (*typeCode == 'B') {
-            // bool-type gets encoded as number number, and doesn't get encoded into 'B', but check anyway
+        
+        if (!strcmp(typeCode, "B") || !strcmp(typeCode, "c")) {
```

Comment:
I'm profoundly confused when it comes to objective C types.. Could you elaborate?
8-bit signed integer do you mean `INT8_C(@1)`? Seems like it was returning a typeCode 'i' correctly for me.
I tried `n.charValue == YES` and it evaluated true with when passing in `@1`...
Will take a look at `kCFBooleanFalse` or `kCFBooleanTrue`, can't seem to make it work for now

Reply:
> I'm profoundly confused when it comes to objective C types

Welcome to the wonderful world of ObjC 😅

Booleans in C/ObjC are often treated as integers. ObjC's `BOOL` type is an alias for `_Bool` (strictly 0 or 1) or `signed char` (-128–127) depending on the architecture you're compiling for. As a result, `NO` and `YES` just mean 0 and 1.

`NSNumber` is a weird type that tries to represent any numbers, including `BOOL` because ObjC treats it like an integer. Regardless of what value `NSNumber` stores, you're going to get some value for `charValue`. Likewise for all the other value getters. `n.charValue == YES` for `@1` makes sense because `@YES` is equivalent to `@1`.

However there exist `kCFBooleanFalse` and `kCFBooleanTrue`, which are `CFBooleanRef` ([bridged to `NSNumber`](https://developer.apple.com/library/archive/documentation/General/Conceptual/CocoaEncyclopedia/Toll-FreeBridgin/Toll-FreeBridgin.html)) instances whose pointer identity indicates true/false. These are the only way to determine true boolean-ness.

To do this check, you can do:

```objc
NSNumber *n = [NSNumber numberWithBool:YES];

if (n == (void*)kCFBooleanFalse || n == (void*)kCFBooleanTrue) {
     // boolean
} else {
     // integer
}
```